Sources/Fuzzilli/Lifting/JavaScriptLifter.swift

```diff
@@ -493,19 +501,26 @@ public class JavaScriptLifter: ComponentBase, Lifter {
             
             if let expression = output {
                 let v = instr.output
-                if policy.shouldInline(expression) && expression.canInline(instr, analyzer.usesIndices(of: v)) {
+                // Do not inline expressions when collecting runtime types, so we have information about all fuzzilli variables
+                if policy.shouldInline(expression) && expression.canInline(instr, analyzer.usesIndices(of: v)) && !options.contains(.collectTypes) {
                     expressions[v] = expression
                 } else {
                     w.emit("\(constDecl) \(v) = \(expression);")
                     if options.contains(.dumpTypes) {
                         w.emitComment("\(v) = \(interpreter.type(of: v))")
+                    } else if options.contains(.collectTypes) {
```

Comment:
Maybe this should just be an `if` instead of an `else if` because it's not documented/obvious that `.dumpTypes` and `.collectTypes` are mutually exclusive?